src/Models/AzureFunctions.ts

```diff
@@ -256,10 +253,21 @@ export class AzureFunctions implements Component, Provisionable, Deployable {
       resourceGroup
     );
     if (functionAppId) {
-      await ConfigHandler.update(ConfigKey.functionAppId, functionAppId);
-      const eventHubConnectionString = ConfigHandler.get<string>(ConfigKey.eventHubConnectionString);
-      const eventHubConnectionPath = ConfigHandler.get<string>(ConfigKey.eventHubConnectionPath);
-      const iotHubConnectionString = ConfigHandler.get<string>(ConfigKey.iotHubConnectionString);
+      const scaffoldType = ScaffoldType.Workspace;
+      const iotHubId = this.dependencies[0].id;
+      const componentConfig = await this.azureConfigFileHandler.getComponentById(scaffoldType, iotHubId);
+      if (!componentConfig) {
+        throw new Error(`Cannot find component with id ${iotHubId}`);
+      }
+      if (!componentConfig.componentInfo) {
+        return false;
+      }
+      const iotHubConnectionString = componentConfig.componentInfo.values.iotHubConnectionString;
+      if (!iotHubConnectionString) {
+        throw new Error("Unable to find Iot Hub connection in the project. Please retry Azure Provision.");
+      }
+      const eventHubConnectionString = componentConfig.componentInfo.values.eventHubConnectionString;
+      const eventHubConnectionPath = componentConfig.componentInfo.values.eventHubConnectionPath;

```

Comment:
It would be better to encapsulate the read/write operation in component level as method of AzureConfigFileHandler utility, and this would be more reusable.